pkg/plugins/autodiscovery/cargo/dependencyManifest.go

```diff
@@ -0,0 +1,81 @@
+package cargo
+
+const (
+	// dependencyManifest is the Go template used to generate
+	// the manifests to update the cargo file
+	dependencyManifest string = `name: '{{ .ManifestName }}'
+sources:
+  {{ .SourceID }}:
+    name: '{{ .SourceName }}'
+    kind: 'cargopackage'
+    spec:
+      package: '{{ .DependencyName }}'
+      versionFilter:
+        kind: '{{ .SourceVersionFilterKind }}'
+        pattern: '{{ .SourceVersionFilterPattern }}'
+{{- if .WithRegistry }}
+      registry:
+        url: '{{ .RegistryURL }}'
+        rootdir: '{{ .RegistryRootDir }}'
+        auth:
+          token: '{{ .RegistryAuthToken }}'
+          headerFormat : '{{ .RegistryHeaderFormat }}'		
+{{- if .RegistrySCMID }}
+    scmid: '{{ .RegistrySCMID }}'
+{{- end }}
+{{- end }}
+  {{ .ExistingSourceID }}:
+    name: '{{ .ExistingSourceName }}'
+    kind: 'toml'
+{{- if .ScmID }}
+    scmid: '{{ .ScmID }}'
+{{- end }}
+    spec:
+      file: '{{ .File }}'
+      Key: '{{ .ExistingSourceKey }}'
+conditions:
+  {{ .ConditionID }}:
+    name: 'Ensure Cargo chart named "{{ .DependencyName }}" is specified'
+    kind: 'toml'
+{{- if .ScmID }}
+    scmid: '{{ .ScmID }}'
+{{- end }}
+    spec:
+      file: '{{ .File }}'
+      query: '{{ .ConditionQuery }}'
+    sourceid: '{{ .ExistingSourceID }}'
+targets:
+  {{ .TargetID }}:
+    name: 'Bump crate dependency "{{ .DependencyName }}" for crate "{{ .CrateName }}"'
+    kind: 'toml'
+{{- if .ScmID }}
+    scmid: '{{ .ScmID }}'
+{{- end }}
+    spec:
+      file: '{{ .TargetFile }}'
+      key: '{{ .TargetKey }}'
+    sourceid: '{{ .SourceID }}'
+{{- if .TargetCargoCleanupEnabled }}
+  Cargo.lock:
+    name: Update Cargo lockfile Cargo.lock
+    dependson:
+      - {{ .TargetID }}
+    disablesourceinput: true
+    kind: shell
+{{- if .ScmID }}
+    scmid: '{{ .ScmID }}'
+{{ end }}
+    spec:
+      command: cargo generate-lockfile
```

Comment:
Is there a way to run this command in dry-run mode?

Reply:
Maybe something like this  cargo update --dry-run $DRY_RUN <package_name>

Reply:
You initial approach by using `cargo generate-lockfile` seems to be the best one.
My only drawback is when running updatecli in diff mode but I don't think there is anything we could do :(